picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -674,16 +673,9 @@ void shutdown() {
    * {@link Picasso} instance. You can either use this directly or by setting it as the global
    * instance with {@link #setSingletonInstance}.
    */
-  public static Picasso with(@NonNull Context context) {
-    if (context == null) {
-      throw new IllegalArgumentException("context == null");
-    }
+  public static Picasso with() {
     if (singleton == null) {
-      synchronized (Picasso.class) {
-        if (singleton == null) {
-          singleton = new Builder(context).build();
-        }
```

Comment:
Actually stash the Context inside a static on the content provider itself. That will prevent this class from initializing until it's referenced.